components/nodemanager-service/nodemanager.go

```diff
@@ -103,14 +135,165 @@ func serve(ctx context.Context, config *config.Nodemanager, connFactory *securec
-	// nodes are assigned to the Automate node manager
-	log.Debug("creating Automate node manager")
+	// Set the cereal service client
+	log.Infof("nodemanager setup, getting a cereal service client %s", config.Cereal.Endpoint)
+	cerealConn, err := connFactory.DialContext(timeoutCtx, "cereal-service", config.Cereal.Endpoint, grpc.WithBlock())
+	if err != nil {
+		err = errors.New("nodemanager setup, error grpc dialing to cereal-service aborting...")
+		return err
+	}
+	// LATER: look at whatever Dan did for picking the domain
+	cerealBackend := grpccereal.NewGrpcBackendFromConn("nodemanager", cerealConn)
+
+	// Set up a cereal (workflow) manager for managing nodemanager workflows.
+	cerealManager, err := cereal.NewManager(cerealBackend)
+	if err != nil {
+		log.Errorf("could not create cereal manager: %s", err)
+	}
+	defer cerealManager.Stop() //nolint:errcheck
+
+	// Prelude to initializing the job manager: register task executors for the nodemanger polling jobs
+	err = cerealManager.RegisterTaskExecutor(Awsec2PollingJobName,
+		&CheckAWSNodesTask{db: db, secretsClient: secretsClient, eventsClient: eventClient},
+		cereal.TaskExecutorOpts{})
+	if err != nil {
+		return err
+	}
+
+	err = cerealManager.RegisterTaskExecutor(AzurevmPollingJobName,
+		&CheckAzureNodesTask{db: db, secretsClient: secretsClient}, cereal.TaskExecutorOpts{})
+	if err != nil {
+		return err
+	}
+
+	err = cerealManager.RegisterTaskExecutor(ManagersPollingJobName,
+		&CheckManagersTask{db: db, secretsClient: secretsClient}, cereal.TaskExecutorOpts{})
+	if err != nil {
+		return err
+	}
+
+	// Prelude to initializing the cereal manager: register workflow executors to run tasks
+	for _, jobName := range []string{Awsec2PollingJobName, AzurevmPollingJobName, ManagersPollingJobName} {
+		err = cerealManager.RegisterWorkflowExecutor(jobName, patterns.NewSingleTaskWorkflowExecutor(jobName, false))
+		if err != nil {
+			return errors.Wrapf(err, "failed to register workflow for %q", jobName)
+		}
+	}
+
+	// SOON: Make all of *waves hands* less repetitive.
+	// Prelude to initializing the manager: create rrules for schedule
+	// Right now, this is migration only -- doesn't handle config change.
+	awsPollInterval := config.Manager.AwsEc2PollIntervalMinutes
+	rule, err := rrule.NewRRule(rrule.ROption{
+		Freq:     rrule.MINUTELY,
+		Interval: awsPollInterval,
+		Dtstart:  time.Now(),
+	})
+	if err != nil {
+		return errors.Wrapf(err, "Could not create recurrence rule for nodemanager workflow %s", Awsec2PollingJobName)
+	}
+
+	azurePollInterval := config.Manager.AzureVMPollIntervalMinutes
+	azureRule, err := rrule.NewRRule(rrule.ROption{
+		Freq:     rrule.MINUTELY,
+		Interval: azurePollInterval,
+		Dtstart:  time.Now(),
+	})
+	if err != nil {
+		return errors.Wrapf(err, "Could not create recurrence rule for nodemanager workflow %s", AzurevmPollingJobName)
+	}
+
+	// managersPollInterval := 120
+	// poll faster for debugging
+	managersPollInterval := 1
```

Comment:
We should get the poll interval from config

Reply:
For consistency, you mean? The current non-cerealized code does not: 
https://github.com/chef/automate/blob/83e0e071be9f124f2fba5a4915e7109c5d7da5cd/components/nodemanager-service/polling/polling.go#L31